### PR TITLE
KokkosKernels/Tpetra stack: local CRS device types

### PR DESCRIPTION
**Pull request by brian-kelley**
Fix the KokkosSparse::CrsMatrix::StaticCrsGraphType specialization to
use the full device type, not just execution space. Was necessary for
some things in KokkosKernels to work with both <Cuda, CudaSpace> an
<Cuda, CudaUVMSpace> in the same build. (Mirror of https://github.com/kokkos/kokkos-kernels/pull/666)

Fix downstream errors in Tpetra, Xpetra and MueLu caused by that change.
This just involved passing the Kokkos::Device specialization
corresponding to Node in to the local_matrix_type/local_graph_type
template parameters.

This is needed by @iyamazaki to integrate the new two-stage GS into Ifpack2.

@trilinos/kokkos-kernels 
@trilinos/tpetra 
@trilinos/muelu 

## Motivation

Just updates Tpetra stack in response to a necessary change to KokkosKernels.

## Related Issues

* Closes 
* Blocks https://github.com/kokkos/kokkos-kernels/pull/671 (merged) getting integrated into Trilinos
* Is blocked by 
* Follows 
* Precedes 
* Related to 
* Part of 
* Composed of https://github.com/kokkos/kokkos-kernels/pull/666

## Stakeholder Feedback

@iyamazaki needs this in order to add two-stage GS for Exawind.
## Testing

Basically just a syntax change - no actual behavior was changed since the Kokkos::Devices used by Tpetra are still always given by an execution space + its default memory space. But, the local graph and local matrix are exercised directly in every CrsGraph/CrsMatrix test.

### CODE REVIEW

**Comment by lucbv** [comment]
If this goes well in the auto-tester we can ask @jjwilke for an approval too!
